### PR TITLE
ADEN-2894 Refactor success/hop flow and allow partner to collapse the slot 

### PR DESCRIPTION
**Pull request by fraszczakszymon**
Done:
- Refactor whole flow of success/hop callbacks - use kind of slot state instead of passing those functions as arguments through many classes
- Add one more partner type (more general) to be able reuse it in future and allow it to collapse slot

Still to do:
ADEN-2930, ADEN-2931, ADEN-2932

### CODE REVIEW

File: extensions/wikia/AdEngine/js/provider/recirculation.js [hunk 1]
```diff
@@ -17,12 +17,12 @@ define('ext.wikia.adEngine.provider.recirculation', [
 		return (properSlot && recirculation);
 	}
 
-	function fillInSlot(slotName, slotElement, success) {
-		log(['fillInSlot', slotName, slotElement], 'debug', logGroup);
+	function fillInSlot(slot) {
+		log(['fillInSlot', slot.name], 'debug', logGroup);
 
-		recirculation.injectRecirculationModule(slotElement);
+		recirculation.injectRecirculationModule(slot.name);
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/recirculation.js]: shouldn't it be `slot.container`?

File: extensions/wikia/AdEngine/js/provider/turtle.js [hunk 2]
```diff
@@ -25,31 +25,27 @@ define('ext.wikia.adEngine.provider.turtle', [
 		return ret;
 	}
 
-	function fillInSlot(slotName, slotElement, success, hop) {
-		log(['fillInSlot', slotName, slotElement, success, hop], 'debug', logGroup);
+	function fillInSlot(slot) {
+		log(['fillInSlot', slot.name], 'debug', logGroup);
 
+		slot.pre('success', function () {
+			var slotName = slot.name;
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/turtle.js]: What are the reasons you decided to assign `slot.name` first to the a variable and re-used it later on?

File: extensions/wikia/AdEngine/js/provider/recirculation.js (hunk 1 above)
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/provider/recirculation.js]: :gem:!

File: extensions/wikia/AdEngine/js/provider/turtle.js (hunk 2 above)
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/provider/turtle.js]: :+1:

**Comment by jacekjursza** [comment]
LGTM

**Pull request merged**